pkg/datapath/iptables/iptables.go

```diff
@@ -404,6 +405,8 @@ var ciliumChains = []customChain{
 
 // IptablesManager manages the iptables-related configuration for Cilium.
 type IptablesManager struct {
+	lock.Mutex
+
```

Comment:
It would be good to document why this lock is needed while `IptablesManager` itself is not modified after `Init()` and why only some of the exported functions take the lock. Apparently this is due to maybe `Install` and `Remove` being called for the same underlying rules concurrently?

Reply:
I'll add a comment :+1: 

> Apparently this is due to maybe Install and Remove being called for the same underlying rules concurrently?

correct, we need this look to ensure there are no multiple instances of `InstallRules()`/`InstallProxyRules()` running concurrently as otherwise we may end up with errors or an inconsistent ruleset